storage/remote/graphite/client.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package graphite
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"net"
+	"net/url"
+	"sort"
+	"time"
+
+	"github.com/prometheus/common/log"
+
+	"github.com/prometheus/common/model"
+)
+
+// Client allows sending batches of Prometheus samples to Graphite.
+type Client struct {
+	url     string
+	timeout time.Duration
+	prefix  string
+}
+
+// NewClient creates a new Client.
+func NewClient(url string, timeout time.Duration, prefix string) *Client {
+	return &Client{
+		url:     url,
+		timeout: timeout,
+		prefix:  prefix,
+	}
+}
+
+func pathFromMetric(m model.Metric, prefix string) string {
+	buffer := bytes.NewBufferString("")
+
+	buffer.WriteString(prefix)
+	buffer.WriteString(escape(m[model.MetricNameLabel]))
+
+	// We want to sort the labels.
+	labels := make(model.LabelNames, len(m))
```

Comment:
You'll want this to be `make(model.LabelNames, 0, len(m))` to create a slice with capacity `len(m)`, but length 0. Otherwise both the length and capacity get set to `len(m)` and the loop below will double that because it appends at the end of that.